### PR TITLE
kedge: Added support for proxying `grpc-web` requests.

### PR DESCRIPTION
**Pull request by Bplotka**
@tonytanger

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/kedge/main.go [hunk 1]
```diff
@@ -69,6 +69,24 @@ var (
...
 )
 
+const (
+	grpcWebContentTypePrefix = "application/grpc-web"
+	grpcContentTypePrefix    = "application/grpc"
+)
+
+func isGRPCReq(header http.Header) bool {
+	// Do not treat "grpc-web" request as pure gRPC. It is pure HTTP/2 at this point.
+	if strings.HasPrefix(header.Get("content-type"), grpcWebContentTypePrefix) {
```
Comment (stefan-improbable) [inline cmd/kedge/main.go]: `content-type` is usually written `Content-Type`. Can you check whether `header.Get` understands this?

File: cmd/kedge/main.go [hunk 2]
```diff
@@ -69,6 +69,24 @@ var (
...
+		return false
+	}
+
+	if strings.HasPrefix(header.Get("content-type"), grpcContentTypePrefix) {
```
Comment (stefan-improbable) [inline cmd/kedge/main.go]: You don't have to use `const` if you use the const string only once. Just put it here directly and it will be easier to read and maintain.

File: cmd/winch/main.go [hunk 3]
```diff
@@ -52,12 +52,12 @@ var (
 		"server_mapper_config",
 		&pb_config.MapperConfig{},
 		"Contents of the Winch Mapper configuration. Content or read from file if _path suffix.").
-		WithFileFlag("default_winch_mapper.json").WithValidator(validateMapper)
+		WithFileFlag("").WithValidator(validateMapper)
```
Comment (stefan-improbable) [inline cmd/winch/main.go]: WHy is this change here?

File: cmd/kedge/main.go (hunk 1 above)
Comment (Bplotka) [reply cmd/kedge/main.go]: Yea it was like that all the time. Works fine.

File: cmd/kedge/main.go (hunk 2 above)
Comment (Bplotka) [reply cmd/kedge/main.go]: If you say so, why not

File: cmd/winch/main.go (hunk 3 above)
Comment (Bplotka) [reply cmd/winch/main.go]: because it is annoying, sorry for that.
Do you really want separate PR for something that small?

File: cmd/kedge/main.go (hunk 2 above)
Comment (stefan-improbable) [reply cmd/kedge/main.go]: We need to minimize amount of LOC to minimum. Less LOC = better readability due to less indirection and less bugs. We can increase LOC to make programs DRY (repeating yourself is worse than more LOC).

File: cmd/winch/main.go (hunk 3 above)
Comment (stefan-improbable) [reply cmd/winch/main.go]: No I just did not know what was this doing and whether this change was intentional :).
